### PR TITLE
[test plan] IPv6 NLRI over IPv4

### PR DESCRIPTION
**Pull request by Azarack**
### Description of PR

Adding test case for IPv6 NLRI over IPv4 feature.

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [X] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
To add new test case.

#### How did you do it?
Full documentation in docs folder.

#### How did you verify/test it?
Verified on T0 vsonic virtual testbed.

#### Any platform specific information?
This is tested with vsonic neighbors only.

#### Supported testbed topology if it's a new test case?
T0 is tested topology.

### Documentation

Yes, docs folder is updated.

### CODE REVIEW

**Comment by yxieca** [comment]
@Azarack can you come to one of the test subgroup meetings and present this test plan?

**Comment by Azarack** [comment]
> @Azarack can you come to one of the test subgroup meetings and present this test plan?

Yes, I have added it to my calendar and our team is planning on attending next week's meeting on Wednesday.

File: tests/bgp/test_ipv6_nlri_over_ipv4.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+from natsort import natsorted
+
+logger = logging.getLogger(__name__)
+# dut_4byte_asn = 400003
+# neighbor_4byte_asn = 400001
+
+
+pytestmark = [
+    pytest.mark.topology('t0'),
+    pytest.mark.device_type('vs')
```
Comment (yxieca) [inline tests/bgp/test_ipv6_nlri_over_ipv4.py:22]: This device type line would live this testcase to run only on vs testbed. You might want to remove this line.

Comment (Azarack) [reply tests/bgp/test_ipv6_nlri_over_ipv4.py]: This has been removed.

**Comment by yxieca** [comment]
@rlhui please assign reviewers. Thanks.

File: tests/bgp/test_ipv6_nlri_over_ipv4.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    cmd = 'vtysh -n {} -c "config" -c "router bgp {}" -c "no neighbor {} peer-group {}" \
+        -c "no neighbor {} peer-group {}"'.format(setup['namespace'], setup['neigh_asn'], setup['dut_ip_v4'],
+                                                  setup['peer_group_v4'], setup['dut_ip_v6'], setup['peer_group_v6'])
+    setup['neighhost'].shell(cmd, module_ignore_errors=True)
+    logger.info("Neighbor BGP Config After Neighbor Removal: {}".format(setup['neighhost']
+                                                                        .shell(cmd="show run bgp")['stdout']))
+    logger.info("Neighbor BGP IPv4 Summary: {}".format(setup['neighhost'].shell(cmd="show ip bgp summary")[u'stdout']))
+    logger.info("Neighbor BGP IPv6 Summary: {}".format(setup['neighhost']
+                                                       .shell(cmd="show ipv6 bgp summary")[u'stdout']))
+    time.sleep(30)
```
Comment (lolyu) [inline tests/bgp/test_ipv6_nlri_over_ipv4.py:133]: Better to wait for the removal by checking the output of `show ip bgp sum` and `show ipv6 bgp sum`(https://github.com/sonic-net/sonic-mgmt/blob/9b1cdee55bbadfc47a54df0ec501869cace17652/tests/common/utilities.py#L104)

File: tests/bgp/test_ipv6_nlri_over_ipv4.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+            .format(setup['neigh_asn'])
+    setup['neighhost'].shell(cmd, module_ignore_errors=True)
+
+    cmd = 'vtysh -c "config" -c "router bgp {}" -c "neighbor {} peer-group NLRI" -c "neighbor {} remote-as {}"\
+        -c "address-family ipv4 unicast" -c "neighbor NLRI activate"'.format(setup['neigh_asn'], setup['dut_ip_v4'],
+                                                                             setup['dut_ip_v4'], setup['dut_asn'])
+    setup['neighhost'].shell(cmd, module_ignore_errors=True)
+    logger.info("Neighbor BGP Config After Peer Config: {}".format(setup['neighhost'].shell('show run bgp')['stdout']))
+
+    time.sleep(30)
```
Comment (lolyu) [inline tests/bgp/test_ipv6_nlri_over_ipv4.py:180]: Same as above, better to wait and check to see if the neighbor is established.

File: tests/bgp/test_ipv6_nlri_over_ipv4.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    # clear BGP table
+    cmd = 'vtysh -n {} -c "clear ip bgp * soft"'.format(setup['namespace'])
+    setup['duthost'].shell(cmd)
+    cmd = 'vtysh -c "clear ip bgp * soft"'
+    setup['neighhost'].shell(cmd)
+
+    # verify route is no longer shared
+    cmd = "show ipv6 route {}".format(setup['dut_nlri_route'])
+    dut_route_out = setup['duthost'].shell(cmd)['stdout']
+    assert setup['neigh_ip_v6'] not in dut_route_out
```
Comment (lolyu) [inline tests/bgp/test_ipv6_nlri_over_ipv4.py:144]: Please use the `pytest_assert`(https://github.com/sonic-net/sonic-mgmt/blob/9b1cdee55bbadfc47a54df0ec501869cace17652/tests/common/helpers/assertions.py#L4)

File: tests/bgp/test_ipv6_nlri_over_ipv4.py (hunk 3 above)
Comment (Azarack) [reply tests/bgp/test_ipv6_nlri_over_ipv4.py]: @lolyu I have made this change.

File: tests/bgp/test_ipv6_nlri_over_ipv4.py (hunk 4 above)
Comment (Azarack) [reply tests/bgp/test_ipv6_nlri_over_ipv4.py]: @lolyu I have made this change.

**Review by lolyu — APPROVED** [review]

**Pull request merged**